ostelco-ios-client/Network/APIManager.swift

```diff
@@ -71,3 +71,34 @@ class APIManager: Service {
         }
     }
 }
+
+extension APIManager {
+    
+    #warning("NSError should contain useful information inside onSuccess")
```

Comment:
Not sure how we should handle the errors inside the onSuccess callback when there is no error to return

Reply:
If we don't have a response object or a region inside that, is that an error?  If it is to be expected, we can return `nil` for error. The user of the method should expect a `nil` region

Reply:
That's one possibility, I was thinking we should construct our own error type, with a descriptive message (maby localizedString?), so that if region is nil, there is always an error and the view does not need to check the error type and decide what text to show, since we decide what to show in the error itself

Reply:
I'll try to add these errors

Reply:
I've added internal error enum inside APIManager for now, until we refactor